### PR TITLE
Disabled option

### PR DESCRIPTION
**Pull request by martyan**
Added support for defining disabled option - option cannot be selected and is greyed out.

### CODE REVIEW

File: src/lib/eventHandlers/handleClick.js [hunk 1]
```diff
@@ -17,13 +17,19 @@ export default function handleClick({ event, state, ReactResponsiveSelectClassRe
       return;
     }
 
+    const optionIndex = parseFloat(event.target.getAttribute('data-key'));
+
+    if (options[optionIndex] && options[optionIndex].disabled === true) {
+      return;
+    }
+
```
Comment (benbowes) [inline src/lib/eventHandlers/handleClick.js]: 👍

**Comment by benbowes** [comment]
Thanks @martyan I appreciate this.

I do have a couple of additions to make
- a couple of tests
- and an aria-disabled attribute

**Comment by benbowes** [comment]
Within the Disabled section of the main tests, something like this... 

```
    it('option can be disabled on multiselect', () => {
      const props = {
        multiselect: true,
        name: 'make',
        options: [{ text: 'Any', value: 'null' }, { text: 'Fiat', value: 'fiat', disabled: true }],
      };
      selectBox = setup(undefined, props);
      expect(selectBox.find('.rrs__option').at(0).props()['aria-disabled']).to.equal('false');
      expect(selectBox.find('.rrs__option').at(1).props()['aria-disabled']).to.equal('true');
      expect(selectBox.find('.rrs__option--disabled').length).to.equal(1);
    });

    it('option can be disabled on singleselect', () => {
      const props = {
        multiselect: true,
        name: 'make',
        options: [{ text: 'Any', value: 'null' }, { text: 'Fiat', value: 'fiat', disabled: true }],
      };
      selectBox = setup(undefined, props);
      expect(selectBox.find('.rrs__option').at(0).props()['aria-disabled']).to.equal('false');
      expect(selectBox.find('.rrs__option').at(1).props()['aria-disabled']).to.equal('true');
      expect(selectBox.find('.rrs__option--disabled').length).to.equal(1);
    });
```

File: src/components/SingleSelectOption.js [hunk 2]
```diff
@@ -28,6 +28,7 @@ export default class SingleSelectOption extends Component {
           rrs__option
           ${(singleSelectSelectedIndex === index) ? 'rrs__option--selected' : ''}
           ${(nextPotentialSelectionIndex === index) ? 'rrs__option--next-selection' : ''}
+          ${(option.disabled === true) ? 'rrs__option--disabled' : ''}
```
Comment (benbowes) [inline src/components/SingleSelectOption.js]: This <li> needs an aria prop now I reckon `aria-disabled={option.disabled ? 'true' : 'false'}`

File: src/components/MultiSelectOption.js [hunk 3]
```diff
@@ -29,6 +29,7 @@ export default class MultiSelectOption extends Component {
           rrs__option
```
Comment (benbowes) [inline src/components/MultiSelectOption.js]: This `<li>` needs an aria prop now I reckon `aria-disabled={option.disabled ? 'true' : 'false'}`

**Comment by benbowes** [comment]
Awesome @martyan looks good to me.

I'll tee up a minor release for 4.1.0.

**Comment by benbowes** [comment]
4.1.0 now published to npm 🎉

**Comment by martyan** [comment]
Thanks for the coop! Great component and neat code)

**Comment by benbowes** [comment]
Thanks heaps for your contribution! I am pretty time poor currently.

**Pull request merged**